Lib/test/test_asyncio/test_futures.py

```diff
@@ -832,6 +832,21 @@ def __eq__(self, other):
 
         fut.remove_done_callback(evil())
 
+    def test_remove_done_callbacks_list_clear(self):
+        # see https://github.com/python/cpython/issues/97592 for details
+
+        fut = self._new_future()
+        fut.add_done_callback(str)
+
+        for _ in range(63):
+            fut.add_done_callback(id)
+
+        class evil:
+            def __eq__(self, other):
+                fut.remove_done_callback(other)
+
+        fut.remove_done_callback(evil())
```

Comment:
All callbacks added by `fut.add_done_callback()` will be cleared when calling `fut.remove_done_callback(Evil())`, but the return value will be the number of callbacks added last.

I'm not sure if we need to improve the return value of `remove done callback` further, the current changes are good enough to fix the crash. :)